src/evaluation/factuality.py

```diff
@@ -97,82 +162,61 @@ def evaluate_factuality(
         "rouge2": [],
         "rougeL": [],
         "sum_with_extrinsic": 0,
+        "count_entity_label": defaultdict(lambda: 0)
     }
-    for value in ANNOTATION_LABELS.values():
-        counters[value] = 0
 
     print_counter = 0
     for sum_id, summary in tqdm(sorted(sums_by_id.items(), key=lambda x: x[0])):
-        source = xsum_test[sum_id]["document"]
-        reference = gold_sums[sum_id]["summary"]
+        summary_eval = evaluate_summary(
+            summary,
+            xsum_test[sum_id]["document"],
+            gold_sums[sum_id]["summary"],
+            labeled_ents[sum_id],
+            is_fbs,
+            is_oracle,
+        )
+        count_non_factual_extrinsic = summary_eval["count_entity_label"][
+            ANNOTATION_LABELS["Non-factual"]
+        ]
+        count_non_factual_intrinsic = summary_eval["count_entity_label"][
+            ANNOTATION_LABELS["Intrinsic"]
+        ]
 
+        # Init the possible eval states of a summary
         non_factual_extrinsic = False
         non_factual_intrinsic = False
-        is_skipped = False
-        has_unknown = False
-        has_failed = False
+        non_factual = False
+        has_unknown = (
+            summary_eval["count_entity_label"][ANNOTATION_LABELS["Unknown"]] > 0
+        )
+        count_extrinsic = 0
 
-        if summary == SUMMARY_FAILED_GENERATION:
+        # Skip logic
+        is_skipped = False
+        if summary_eval["failed"]:
+            is_skipped = True
+        elif is_fbs and is_oracle and count_non_factual_extrinsic:
+            is_skipped = True
+        elif is_fbs and summary_eval["has_predicted_non_factual"]:
             is_skipped = True
-            has_failed = True
-            counters["failed"] += 1
-        else:
-            rouge_scores = rouge([summary], [reference])
-            counters["rouge1"].append(rouge_scores["rouge1"]["f1"])
-            counters["rouge2"].append(rouge_scores["rouge2"]["f1"])
-            counters["rougeL"].append(rouge_scores["rougeL"]["f1"])
 
-        n_extrinsic = 0
-        for ent in labeled_ents[sum_id]:
-            if is_gold:
-                if ent["in_source"]:
-                    counters[ANNOTATION_LABELS["Factual"]] += 1
-                else:
-                    counters[ANNOTATION_LABELS["Non-hallucinated"]] += 1
-                counters["entities"] += 1
-                continue
-
-            # SKIP LOGIC for FBS
-            if is_fbs:
-                is_oracle = "predicted_label" not in ent
```

Comment:
was `is_oracle` always a misnomer? Why the change?

Reply:
The logic of this method has been completely refactored so hard to explain why this line changed - unsure what you mean by misnomer.

The `is_oracle` flag was correct, but has been moved to the function args and is no longer derived from "predicted_label".